code/modules/robotics/bot/duckbot.dm

```diff
@@ -51,44 +51,24 @@
 /obj/machinery/bot/duckbot/proc/wakka_wakka()
 	if(moving) return
 	if(src.emagged)
-		if(src.annoy_target)
-			if(!ON_COOLDOWN(src, DUCKBOT_ANNOY_TIMEOUT, src.quack_cooldown))
+		if(ismob(src.annoy_target))
+			if(!GET_COOLDOWN(src, DUCKBOT_ANNOY_TIMEOUT))
 				src.KillPathAndGiveUp(1)
-			else if(!ON_COOLDOWN(src, DUCKBOT_ANNOY_PATHING_COOLDOWN, src.annoy_path_cooldown))
+			else if(!GET_COOLDOWN(src, DUCKBOT_ANNOY_PATHING_COOLDOWN))
+				ON_COOLDOWN(src, DUCKBOT_ANNOY_PATHING_COOLDOWN, src.annoy_path_cooldown)
 				var/turf/randwander = get_step_rand(get_turf(src.annoy_target))
-				src.navigate_to(randwander, DUCKBOT_MOVE_SPEED, 1, 30)
+				src.navigate_to(randwander, DUCKBOT_MOVE_SPEED, 0, 30)
 		else
 			for_by_tcl(M, /mob)
 				if(IN_RANGE(src, M, 7))
-					if(M in src.annoyed_nerds)
-						if(!ON_COOLDOWN(src, "[DUCKBOT_ANNOY_LOCKOUT_TIMEOUT]-[M.name]", src.forget_annoyed_timeout))
-							src.annoyed_nerds -= M
-						else
-							continue
-					ON_COOLDOWN(src, DUCKBOT_ANNOY_TIMEOUT, src.quack_cooldown)
-					ON_COOLDOWN(src, "[DUCKBOT_ANNOY_LOCKOUT_TIMEOUT]-[M.name]", src.forget_annoyed_timeout)
-					src.annoy_target = M
-					src.annoyed_nerds[M] = TRUE
-					src.navigate_to(M, DUCKBOT_MOVE_SPEED, 1, 30)
-					break
+					if(!GET_COOLDOWN(src, "[DUCKBOT_ANNOY_LOCKOUT_TIMEOUT]-[M.name]"))
+						ON_COOLDOWN(src, "[DUCKBOT_ANNOY_LOCKOUT_TIMEOUT]-[M.name]", src.forget_annoyed_timeout)
```

Comment:
why not on cooldown

Reply:
I thought that was messing with it, but I forgot to change it back after I found the real problem. Fixed!